src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCodeExecutor.java

```diff
@@ -165,19 +165,24 @@ public void onEvent(SparkListenerEvent event) {
     }
   }
 
+  public boolean isDataFrame() {
+    return isDataFrame;
+  }
+
   /**
    * Execute interpreted code on the given RDD.
    */
   public Object execute(SparkExecutionPluginContext context,
                         JavaRDD<StructuredRecord> javaRDD) throws InvocationTargetException, IllegalAccessException {
     // RDD case
     if (!isDataFrame) {
+      Object result;
       if (takeContext) {
         //noinspection unchecked
-        return ((RDD<StructuredRecord>) method.invoke(null, javaRDD.rdd(), context)).toJavaRDD();
+        return method.invoke(null, javaRDD.rdd(), context);
       } else {
         //noinspection unchecked
```

Comment:
do you still need that now that you are not casting?

Reply:
good catch, will remove